plugins/dereferenceUses.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+
+const { querySelectorAll, querySelector } = require('../lib/xast.js');
+
+exports.type = 'full';
+
+exports.active = true;
+
+exports.description = 'dereferences <use/> elements';
+
+exports.params = {
+  keepHref: false, // keep (xlink:)href attributes
+
+  symbolContainer: 'svg', // browsers use <svg/> as container of <symbol/> content (`g` could also be used)
+};
+
+const OverridingUseAttributeNames = [
+  'x',
+  'y',
+  'width',
+  'height',
+  'href',
+  'xlink:href',
+];
+const HrefAttributeNames = ['href', 'xlink:href'];
+
+/**
+ * Dereferences <use> elements
+ *
+ *
+ * @param {Object} document document element
+ * @param {Object} options plugin params
+ *
+ * @author strarsis <strarsis@gmail.com>
+ */
+exports.fn = function (document, options) {
+  options = options || {};
```

Comment:
Call it params as everywhere please. Btw, options is always object. And use destructuring to set defaults please
```js
const {
  keepHref = false,
  symbolContainer = 'svg',
} = params;
```

Reply:
Done.